### PR TITLE
Pass in strong password instead to fix integtest workflow

### PR DESCRIPTION
**Pull request by derek-ho**
### Description
With recent changes in security plugin requiring a strong initial admin password to be set, I've seen a few autocuts with integtest failures. This PR attempts to resolve those by passing in a strong password and using that to run the tests

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by derek-ho** [comment]
Am I correct in assuming that we also need to change any repo which has its own integtest.sh file to use default admin:myStrongPassword123! with this change? How do we handle cases in which integtest pass in its own credentials, or we don't need to worry about those?

File: src/test_workflow/integ_test/distribution_deb.py
Comment (rishabh6788) [inline src/test_workflow/integ_test/distribution_deb.py:35]: This will only set the env variable for the duration of the `sudo env` command execution and will not persist. Recommend to use `export` to set the environment variable.

File: src/test_workflow/integ_test/distribution_rpm.py
Comment (rishabh6788) [inline src/test_workflow/integ_test/distribution_rpm.py:35]: same as above.

File: src/test_workflow/integ_test/distribution_tar.py
Comment (rishabh6788) [inline src/test_workflow/integ_test/distribution_tar.py:36]: I believe this should be `export OPENSEARCH_INITIAL_ADMIN_PASSWORD=myStrongPassword123! && ./opensearch-tar-install.sh`

**Comment by rishabh6788** [comment]
Added a few comments, please address them. 
Also please fix the failing tests.

File: scripts/default/integtest.sh
Comment (rishabh6788) [inline scripts/default/integtest.sh:92]: This wouldn't work for OS version lower than 2.12. 
It may not make much sense to run integration tests for released 2.x versions but it will definitely break integ tests for 1.x line.

Comment (gaiksaya) [reply scripts/default/integtest.sh:92]: Right! The code is generic for all versions as of today. Until we solve the branching problem.

Comment (prudhvigodithi) [reply scripts/default/integtest.sh:92]: To unblock this we can try adding if/else condition to differentiate with 1.x and above until we have this branching/versioning for the build repo solved. The script already has an argument `OPENSEARCH_VERSION` so adding if/else should be straight forward.

Comment (derek-ho) [reply scripts/default/integtest.sh:92]: I think issue with that is what if patch version 2.0 < patch version < 2.12 come along? I was looking into bash script to compare semantic version, but it seems complex, not sure how do you folks feel about adding something like this: https://stackoverflow.com/questions/4023830/how-to-compare-two-strings-in-dot-separated-version-format-in-bash?

Comment (derek-ho) [reply scripts/default/integtest.sh]: @rishabh6788 @gaiksaya @prudhvigodithi I changed the logic in the script itself to set a strong password if the version is after 2.12.0. Let me know if that makes sense to keep. I also included this logic in the integtest.sh files of repo's which have their own versions of integtest.sh.

File: scripts/default/integtest.sh
Comment (peterzhuamazon) [inline scripts/default/integtest.sh:100]: 1. You are changing IFS without resetting it back to original
2. Why not use awk or cut with separator like on line 95/103?

**Review by peterzhuamazon — CHANGES_REQUESTED** [review]
This is just the default integTest.sh, there are more in each plugin's repo.

**Comment by derek-ho** [comment]
> This is just the default integTest.sh, there are more in each plugin's repo.

I believe most plugins have removed their custom integTest.sh in favor of this default. There are two repos I found which need their own custom ones, we have open PRs in those, which should be merged prior to 2.12.0 release, correct me if my understanding is incorrect:
https://github.com/opensearch-project/cross-cluster-replication/pull/1298
https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/981
If you can take a look at these, that would also be awesome! I do not know if their repo's branching strategy is honored. @kavilla suggested we don't need logic in these repos to determine version since those repo's integtest.sh is run against their own branch.

File: scripts/default/integtest.sh
Comment (derek-ho) [reply scripts/default/integtest.sh]: Fixed as discussed with this commit: [7480dde](https://github.com/opensearch-project/opensearch-build/pull/4302/commits/7480dde2a05f95ac25de7c9644b7197239e1875f)

**Review by DarshitChanpura — APPROVED** [review]

**Review by peterzhuamazon — COMMENTED** [review]
Waiting for final decision based on this issue before approval:
https://github.com/opensearch-project/security/issues/3916

**Review by prudhvigodithi — APPROVED** [review]

File: src/test_workflow/integ_test/distribution_tar.py
Comment (prudhvigodithi) [inline src/test_workflow/integ_test/distribution_tar.py:36]: FYI, the `export OPENSEARCH_INITIAL_ADMIN_PASSWORD=myStrongPassword123!` would be redundant for versions below 2.12.0, but should not block the installation.

**Review by peterzhuamazon — APPROVED** [review]

**Comment by peterzhuamazon** [comment]
You can either add docker or open a new PR for docker:
https://github.com/opensearch-project/opensearch-build/blob/main/docker/release/config/opensearch/opensearch-docker-entrypoint.sh#L33-L39

**Comment by derek-ho** [comment]
> You can either add docker or open a new PR for docker: https://github.com/opensearch-project/opensearch-build/blob/main/docker/release/config/opensearch/opensearch-docker-entrypoint.sh#L33-L39

Is this guy being used in docker integtest? Can you point me to where docker integtest is run, if at all? I would rather merge in this PR as is, since the main purpose is to sniff out if this will solve the autocuts that have been cut against a lot of repos, and close any false alarm that is being raised.

**Comment by prudhvigodithi** [comment]
@derek-ho Merging the PR to make sure 2.12.0 `with-security` integ tests pass, please keep an eye on failures if there is anything breaking with this change.

**Comment by DarshitChanpura** [comment]
> You can either add docker or open a new PR for docker:
https://github.com/opensearch-project/opensearch-build/blob/main/docker/release/config/opensearch/opensearch-docker-entrypoint.sh#L33-L39

This is addressed in this PR: https://github.com/opensearch-project/opensearch-build/pull/4274

**Pull request merged**
